### ISSUE TITLE
strict = True does not correctly override parent settings

### ISSUE THREAD

**Issue by TylerYep** [python/mypy#14372]
**Bug Report**

**To Reproduce**
I use the following setup.cfg:

```
[mypy-apps.*]
allow_untyped_defs = True
allow_incomplete_defs = True

[mypy-apps.new_feature.*]
strict = True
```

```python
def hello(x) -> None:
     return
```

**Actual Behavior**

```
Success: no issues found in 194 source files.
```

**Expected Behavior**

I expected `strict=True` to override the parent apps.* settings and raise the untyped def error, but it did not unless I manually set the attributes again via:

```
[mypy-apps.new_feature.*]
ignore_errors = False
strict = True
allow_untyped_calls = True
warn_return_any = False
allow_untyped_defs = False
allow_incomplete_defs = False
```

```
apps/new_feature/hello.py:7: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
```

**Your Environment**

- Mypy version used: mypy 0.991 and 0.812
- Mypy command-line flags: `mypy .`
- Mypy configuration options from `mypy.ini` (and other config files): See above
- Python version used: 3.11 and 3.8

**Comment by hauntsaninja** [comment]
Duplicate of https://github.com/python/mypy/issues/11401
